lib/utils/pyexec.c

```diff
@@ -127,6 +131,10 @@ STATIC int parse_compile_execute(const void *source, mp_parse_input_kind_t input
         start = mp_hal_ticks_ms();
         #endif
         mp_call_function_0(module_fun);
+        // execute exit handlers (if any).
+        #if CIRCUITPY_ATEXIT
+        shared_module_atexit_execute();
```

Comment:
I think you'll need to call this in the REPL too. This is only used for executing a file I think...

Reply:
Done.